tasks/README.md

```diff
@@ -104,8 +109,28 @@ If a parameter type accepts `null` the task runner will accept either a `null` v
 
 ### Metaparameters
 
-In addition to the tasks parameters, the task runner may inject metaparameters prefixed by '_'. These include `_noop` and `_task`.
+In addition to the tasks parameters, the task runner may inject metaparameters prefixed by `_`.
+
+**_noop**: Used to implement logic in tasks to support cases where task should not perform certain actions.
+
+**_task**: Allow multiple task implementations to access the same executable file. The `_task` metaparameter provides the executable the task name to allow task specific logic to be implemented. 
+
+**_installdir**: Tasks with `files` specified in the metatdata will be passed the `_installdir` metaparam to provide the file path to the expected executables. 
 
+#### Metaparameter Example
+
+When the task runner runs a task with files it copies those files into the same temporary directory as the task. The task can then load the code based on the path. The directory structure of the specified file resources will be preserved such that paths specified with the `files` metatdata option will be available to tasks prefixed with `_installdir`. 
```

Comment:
For this example we assume that the directory structure is preserved when resource is provided to task via the `_installdir` metaparameter. Do we want to consider other options such as flattening directory structure when copying `files` resources to targets?

Reply:
No, we should always preserve the directory structure.